### PR TITLE
task_DEV-1

### PR DESCRIPTION
**Pull request by KatsiarynaBukraba**

### CODE REVIEW

File: DuplicateSymbols.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+﻿using System;
+
+namespace LineAnalysis
+{
+    /// <summary>
```
Comment (IgorKheidorov) [inline DuplicateSymbols.cs]: Class can't be "about", bad comment

File: DuplicateSymbols.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                    {
+                        maxCount = count;
+                    }
+                    count = 0;
+                }
+
+            }
+
+            return maxCount + 1;
+
```
Comment (IgorKheidorov) [inline DuplicateSymbols.cs]: empty line

File: EntryPoint.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    /// </summary>
+    class EntryPoint
+    {
+        /// <summary>
+        /// It's an entry point in the program.
+        /// </summary>
+        /// <param name="args">It's the users string to input.</param>
+        static void Main(string[] args)
+        {
+            DuplicateSymbols userString = new DuplicateSymbols(args[0]);
```
Comment (IgorKheidorov) [inline EntryPoint.cs]: bad value name

File: DuplicateSymbols.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                    count++;
+                }
+                else
+                {
+                    if(count > maxCount)
+                    {
+                        maxCount = count;
+                    }
+                    count = 0;
+                }
```
Comment (IgorKheidorov) [inline DuplicateSymbols.cs]: empty string

File: DuplicateSymbols.cs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+            }
+
+            return maxCount + 1;
```
Comment (IgorKheidorov) [inline DuplicateSymbols.cs]: Why +1? better int count = 1

File: EntryPoint.cs [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    /// It's an entry point in the program.
+    /// </summary>
+    class EntryPoint
+    {
+        /// <summary>
+        /// It's an entry point in the program.
+        /// </summary>
+        /// <param name="args">It's the users string to input.</param>
+        static void Main(string[] args)
+        {
```
Comment (IgorKheidorov) [inline EntryPoint.cs]: Check number of arguments

File: DuplicateSymbols.cs [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    {
+        private string userString;
+
+        public DuplicateSymbols(string args)
+        {
+            userString = args;
+        }
+
+        /// <summary>
+        /// This method researches users string to found maximal series of duplicate symbols.
```
Comment (IgorKheidorov) [inline DuplicateSymbols.cs]: "This method researches..." bad comment, method do not research, it find or calculate

**Review by IgorKheidorov — CHANGES_REQUESTED** [review]

File: LineAnalysis/DuplicateSymbols.cs [hunk 8]
```diff
@@ -0,0 +1,44 @@
+﻿using System;
+
+namespace LineAnalysis
+{
+    /// <summary>
+    ///  This class contains methods for work with duplicate symbols.
```
Comment (IgorKheidorov) [inline LineAnalysis/DuplicateSymbols.cs]: duplicated

File: LineAnalysis/DuplicateSymbols.cs [hunk 9]
```diff
@@ -0,0 +1,44 @@
+﻿using System;
+
+namespace LineAnalysis
+{
+    /// <summary>
+    ///  This class contains methods for work with duplicate symbols.
+    /// </summary>
+    class DuplicateSymbols
```
Comment (IgorKheidorov) [inline LineAnalysis/DuplicateSymbols.cs]: Better DuplicateSymbolsCounter

File: LineAnalysis/DuplicateSymbols.cs [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+
+        /// <summary>
+        /// This method calculates the maximal series of duplicate symbols of user's string.
+        /// </summary>
+        /// <returns>maximal length of the series of duplicate symbols</returns>
+        public int MaxSerialDuplicateSymbols()
+        {
+            int currentCount = 1;
+            int maxCount = 1;
+
```
Comment (IgorKheidorov) [inline LineAnalysis/DuplicateSymbols.cs]: Ha- Ha, its bug, not correct result returns in one case, look for

**Review by IgorKheidorov — CHANGES_REQUESTED** [review]

File: LineAnalysis/DuplicateSymbolsCounter.cs [hunk 11]
```diff
@@ -0,0 +1,44 @@
+﻿using System;
+
+namespace LineAnalysis
+{
+    /// <summary>
+    ///  This class contains methods for work with duplicate symbols.
+    /// </summary>
+    class DuplicateSymbolsCounter
+    {
+        private string UserString { get; set; }
```
Comment (AlexanderSoroka) [inline LineAnalysis/DuplicateSymbolsCounter.cs]: bad property name - who guarantee that string always relates to user?

File: LineAnalysis/DuplicateSymbolsCounter.cs [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+namespace LineAnalysis
+{
+    /// <summary>
+    ///  This class contains methods for work with duplicate symbols.
+    /// </summary>
+    class DuplicateSymbolsCounter
+    {
+        private string UserString { get; set; }
+
+        public DuplicateSymbolsCounter(string args)
```
Comment (AlexanderSoroka) [inline LineAnalysis/DuplicateSymbolsCounter.cs]: bad argument name, "value" is much better

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]
readme.txt doesn't relate to the task, remove it from review

File: LineAnalysis/DuplicateSymbolsCounter.cs [hunk 13]
```diff
@@ -0,0 +1,44 @@
+using System;
+
+namespace LineAnalysis
+{
+    /// <summary>
+    ///  This class contains methods for work with duplicate symbols.
+    /// </summary>
```
Comment (IgorKheidorov) [inline LineAnalysis/DuplicateSymbolsCounter.cs]: DublicatedSymbols..

File: LineAnalysis/DuplicateSymbolsCounter.cs [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+    class DuplicateSymbolsCounter
+    {
+        private string InputString { get; set; }
+
+        public DuplicateSymbolsCounter(string value)
+        {
+            InputString = value;
+        }
+
+        /// <summary>
```
Comment (IgorKheidorov) [inline LineAnalysis/DuplicateSymbolsCounter.cs:17]: Dublicated

**Review by IgorKheidorov — APPROVED** [review]

**Pull request merged**